### PR TITLE
Add sorting to the loot tracker

### PR DESCRIPTION
**Pull request by Eingin**
Addresses: #8519 

Adds a sorting setting into the loot tracker config that lets users control how the drop group entries are sorted.

Currently supports:
- Most recent drop
- Lest recent drop
- Highest value
- Lowest value

Config Panel

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootEntrySorting.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
+package net.runelite.client.plugins.loottracker;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootEntrySorting.java]: License header

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootEntrySorting.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+import lombok.Getter;
+
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootEntrySorting.java]: Fix your imports

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootEntrySorting.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+		(LootTrackerBox a, LootTrackerBox b) ->
+			Long.compare(b.getTotalPrice(), a.getTotalPrice())
+	);
+
+
+	private final String name;
+
+	@Getter
+	private final Comparator<LootTrackerBox> sorter;
+
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootEntrySorting.java]: remove this empty line

**Comment by Eingin** [comment]
@deathbeam Opps, that should fix it all up

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootEntrySorting.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.loottracker;
+
+import lombok.Getter;
+import java.util.Comparator;
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootEntrySorting.java]: Imports should be listed alphabetically.

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPanel.java [hunk 5]
```diff
@@ -341,8 +341,7 @@ void add(final String eventName, final int actorLevel, LootTrackerItem[] items)
 		LootTrackerBox box = buildBox(record);
 		if (box != null)
 		{
-			box.rebuild();
-			updateOverall();
+			rebuild();
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPanel.java]: This seems much more expensive now, we've had issues in the past with perf. issues on the loottracker due to too much rebuilding. @abextm do you know if this is ok?

Comment (abextm) [reply runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPanel.java]: Yeah, doing this will reopen #6134

**Pull request closed without merging**